api/src/Repository/DepartmentRepository.php

```diff
@@ -116,7 +117,23 @@ public function update(/*.string.*/$id, /*.mixed.*/$data): void
 
     public function deleteById(/*.mixed.*/$id): void
     {
-        throw new Exception(__CLASS__.": Method '__FUNCTION__' not implemented");
+        $placeholderDeptId = Select::new($this->db)
+            ->columns("DepartmentID")
+            ->from("Departments")
+            ->where("Name = 'Historical Placeholder'")
+            ->fetchOne();
+        
+        $update = Update::new($this->db);
+        $update->table("Departments")
+            ->column("FallbackID", null)
+            ->column("ParentDepartmentID", $placeholderDeptId["DepartmentID"])
+            ->whereEquals(["DepartmentID" => $id])
+            ->perform();
+
+        $delete = Delete::new($this->db);
+        $delete->from("Departments")
+            ->whereEquals(["DepartmentID" => $id])
+            ->perform();
 
```

Comment:
So deleting departments has been a bit tricky.  This is because it breaks historical concom records and nobody is happy about that. 

if you look at `modules/concom/functions/concom.inc` and `delete_position` around line 336 you will see that we do not delete a position (department)  instead we set the Parent to be the magical 'Historical Placeholder' department.  That marks it in the system as a department that is no long active.  But it keeps all the dependancies in the concom records so that we do not loose any of the years served sort of logic.

Reply:
Sure, I can make that adjustment instead and opt to simply update it to use the historical as the parent.

It sounds like years served on ConCom is the only major requirement. If we had another approach for that would we be able to delete the departments? Or is the preference to keep with this forever?

Reply:
When I initially implemented this I asked if we wanted to track just years served or years served with what department and I never was given a clear answer.  Especially since they have a tendency to rename departments a lot, which already breaks that sort of history in some ways. But I just errored with the way that saved the most data possible for now.  It does mean the convention has many many historical departments, which is not a lot of data but still gets a bit messy feeling.